### PR TITLE
TextEditor: Added GML preview checkbox under 'View' and GML Preview

### PR DESCRIPTION
**Pull request by nanoproductions**
Added the ability to view GML files in the TextEditor and work on them!

### CODE REVIEW

**Comment by danboid** [comment]
Sounds v.cool!

File: Applications/TextEditor/TextEditorWidget.cpp [hunk 1]
```diff
@@ -651,6 +668,13 @@ void TextEditorWidget::update_html_preview()
     m_page_view->scroll_into_view(current_scroll_pos, true, true);
 }
 
+void TextEditorWidget::update_GML_preview()
+{
+    auto current_scroll_pos = m_page_view->visible_content_rect();
+    m_page_view->load_from_gml(m_editor->text());
```
Comment (awesomekling) [inline Applications/TextEditor/TextEditorWidget.cpp:674]: `m_page_view` is a web view, we should not be inserting random children into it.
This should be done with an empty `GUI::Widget` at the same level as `m_page_view`, possibly with both in a `GUI::StackWidget`.

File: Applications/TextEditor/TextEditorWidget.h [hunk 2]
```diff
@@ -61,6 +62,7 @@ class TextEditorWidget final : public GUI::Widget {
     void update_preview();
     void update_markdown_preview();
     void update_html_preview();
+    void update_GML_preview();
```
Comment (awesomekling) [inline Applications/TextEditor/TextEditorWidget.h:65]: `update_gml_preview()`

**Review by awesomekling — CHANGES_REQUESTED** [review]

**Comment by awesomekling** [comment]
I don't believe this feature is useful, since we already have a dedicated program (GML Playground) for editing GML. This seems like it would only result in having two instances of the same functionality, and this one always being weaker than what's available in GML Playground.

**Pull request closed without merging**